common/functions.js

```diff
@@ -0,0 +1,179 @@
+            }
+        })
+
+        body.addEventListener('click', (event) => {
+            if(!event.target.closest('.mobile-menu_open') && menu.classList.contains('mobile-menu_open')) {
+                setMobileStyles();
+            }
+        })
+
+
+        function setMobileStyles() {
+            buttonBurger.classList.toggle('rotate');
+            layout.classList.toggle('layout_mobile');
+            content.classList.toggle('layout_mobile');
+            body.classList.toggle('no-scroll');
+            backgroundWrapper.classList.toggle('is-open');
+
+            if(menu.classList.contains('mobile-menu_close')) {
+                menu.classList.add('mobile-menu_open');
+                menu.classList.remove('mobile-menu_close');
+            } else {
+                menu.classList.add('mobile-menu_close');
+                menu.classList.remove('mobile-menu_open');
+            }
+        }
+    }
+
+
+    function createCards(arr, parentElem) {
+        parentElem.textContent = ' ';
+
+        for (let i = 0; i < arr.length; i++) {
+            const cardPets = document.createElement('article');
+            cardPets.classList.add('card');
+
+            const petName = document.createElement('h4');
+            const petPhoto = document.createElement('img');
+
+            const wrapper = document.createElement('div');
+            wrapper.classList.add('wrapper-column');
+
+            const cardButton = document.createElement('button');
+            cardButton.innerText = 'Learn more';
+            cardButton.className = 'button';
+
+            petName.innerText = arr[i].pet.name;
+            petPhoto.setAttribute('src', arr[i].pet.img);
+            cardPets.appendChild(petPhoto);
+            cardPets.appendChild(petName);
+            cardPets.appendChild(cardButton);
+            parentElem.appendChild(cardPets);
+
+            cardPets.addEventListener('click',(event) =>  {
+                const body = document.querySelector('body');
+                body.classList.add('no-scroll');
+                popup(arr[i].pet);
+                event.stopPropagation();
+            });
+        }
+    }
+
+
+    function adaptDataForSliders (arr) {
+        let copy  = JSON.parse(JSON.stringify(arr));
+        return copy.map((item) => {
+            return {pet:item,
+                    id: `${item.name}.${item.type}.${item.age}`
+            }
+        })
+    }
+
+
+    function getRandomPets(data, cardsCount, currentPetsArr = []) {
+        let copyData = JSON.parse(JSON.stringify(data));
+        let nextPets = [];
+        let dataForGettingRandomPets;
+        if(cardsCount > data.length) {
+            console.error('getRandomPets: cardsCount is incorrect');
+        } else {
+            /* conditionals for creating and comparing next cards */
+            if (currentPetsArr.length !== 0 && currentPetsArr.length !== data.length )  {
+                /* creating next cards without repeated pets */
```

Comment:
Please, Try to avoid nesting code too many levels deep. https://javascript.info/coding-style, https://learn.javascript.ru/coding-style

Reply:
Thanks for materials. I will fix code style. I afraid I can`t to avoid deep levels in this functional because I need conditional rendering and considering edge values